model_analyzer/model_manager.py

```diff
@@ -55,6 +56,13 @@ def __init__(self, config, gpus, client, server, metrics_manager,
         self._result_manager = result_manager
         self._state_manager = state_manager
 
+        if state_manager.starting_fresh_run():
+            self._init_state()
+
+        self._model_variant_name_manager = ModelVariantNameManager.from_dict(
```

Comment:
Is there something else that should handle this?  It seems wrong for model manager to need to get MVNM and state_manager to interact in such a precise way.

Reply:
We've talked about this coupling in the past and I agree it's awkward, but I don't of any other way to do this with how things currently exist. it's beyond the scope of this story to make a change, and I'm not sure it's worth the $/! with so few classes needing checkpoints.